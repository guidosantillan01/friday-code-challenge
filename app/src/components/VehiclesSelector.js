import React from 'react';

const selector = (onChangeHandler, options) => {
  return (
    <div>
      <select onChange={onChangeHandler} className="v-selector">
        {options.map(option => {
          return (
            <option value={option} key={option}>
              {option}
            </option>
          );
        })}
      </select>
    </div>
  );
};

const VehicleSelector = ({
  makes,
  models,
  selectVehicleMake,
  selectVehicleModel
}) => {
  return (
    <div>
      <h2 className="App__subtitle">Selector for Vehicles</h2>

      <div className="v-selector__container">
        {selector(selectVehicleMake, makes)}
        {selector(selectVehicleModel, models)}
      </div>
    </div>
  );
};

export default VehicleSelector;
