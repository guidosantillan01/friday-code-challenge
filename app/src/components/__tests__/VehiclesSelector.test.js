import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';

import VehiclesSelector from '../VehiclesSelector';

const makes = ['BWM', 'Ford', 'Tesla'];
const models = ['Model S', 'Model 3', 'Model X', 'Model Y'];
const selectVehicleMake = () => {};
const selectVehicleModel = () => {};

describe('Vehicle Selector', () => {
  afterEach(cleanup);

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <VehiclesSelector
        makes={makes}
        models={models}
        selectVehicleMake={selectVehicleMake()}
        selectVehicleModel={selectVehicleModel()}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  test('should render VehiclesSelector correctly', () => {
    const { container } = render(
      <VehiclesSelector
        makes={makes}
        models={models}
        selectVehicleMake={selectVehicleMake()}
        selectVehicleModel={selectVehicleModel()}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
