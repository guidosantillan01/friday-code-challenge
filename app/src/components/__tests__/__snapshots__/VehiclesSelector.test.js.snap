// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicle Selector should render VehiclesSelector correctly 1`] = `
<div>
  <div>
    <h2
      class="App__subtitle"
    >
      Vehicle Selector
    </h2>
    <div
      class="v-selector__container"
    >
      <div>
        <select
          class="v-selector"
        >
          <option
            value="BWM"
          >
            BWM
          </option>
          <option
            value="Ford"
          >
            Ford
          </option>
          <option
            value="Tesla"
          >
            Tesla
          </option>
        </select>
      </div>
      <div>
        <select
          class="v-selector"
        >
          <option
            value="Model S"
          >
            Model S
          </option>
          <option
            value="Model 3"
          >
            Model 3
          </option>
          <option
            value="Model X"
          >
            Model X
          </option>
          <option
            value="Model Y"
          >
            Model Y
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
